### PR TITLE
adding Program Group and Platform Model into the Read Group Record after

### PR DESCRIPTION
**Pull request by nh13**
the update to the SAM spec (samtools/hts-specs#16).

### CODE REVIEW

**Comment by nh13** [comment]
See #188.

File: src/java/htsjdk/samtools/SAMReadGroupRecord.java [hunk 1]
```diff
@@ -39,21 +39,24 @@
...
-    public static final String PLATFORM_UNIT_TAG = "PU";
-    public static final String SEQUENCING_CENTER_TAG = "CN";
-    public static final String PLATFORM_TAG = "PL";
     public static final String LIBRARY_TAG = "LB";
+    public static final String PROGRAM_GROUP = "PG";
+    public static final String PREDICTED_MEDIAN_INSERT_SIZE_TAG = "PI";
+    public static final String PLATFORM_TAG = "PL";
+    public static final String PLATFORM_MODEL = "PM";
+    public static final String PLATFORM_UNIT_TAG = "PU";
+    public static final String READ_GROUP_SAMPLE_TAG = "SM";
```
Comment (nh13) [inline src/java/htsjdk/samtools/SAMReadGroupRecord.java]: Changed the order to match the same spec

File: src/java/htsjdk/samtools/SAMReadGroupRecord.java [hunk 2]
```diff
@@ -39,21 +39,24 @@
...
 
     public static final Set<String> STANDARD_TAGS =
-            new HashSet<String>(Arrays.asList(READ_GROUP_ID_TAG, READ_GROUP_SAMPLE_TAG, LIBRARY_TAG,
-        DESCRIPTION_TAG, PLATFORM_UNIT_TAG, PREDICTED_MEDIAN_INSERT_SIZE_TAG, SEQUENCING_CENTER_TAG,
-            DATE_RUN_PRODUCED_TAG, PLATFORM_TAG, FLOW_ORDER_TAG, KEY_SEQUENCE_TAG));
+            new HashSet<String>(Arrays.asList(READ_GROUP_ID_TAG, SEQUENCING_CENTER_TAG, DESCRIPTION_TAG,
+                    DATE_RUN_PRODUCED_TAG, FLOW_ORDER_TAG, KEY_SEQUENCE_TAG, LIBRARY_TAG,
+                    PROGRAM_GROUP, PREDICTED_MEDIAN_INSERT_SIZE_TAG, PLATFORM_TAG, PLATFORM_MODEL,
+                    PLATFORM_UNIT_TAG, READ_GROUP_SAMPLE_TAG));
```
Comment (nh13) [inline src/java/htsjdk/samtools/SAMReadGroupRecord.java]: Changed the order to match the order in the SAM spec.

File: src/java/htsjdk/samtools/SAMReadGroupRecord.java [hunk 3]
```diff
@@ -39,21 +39,24 @@
...
+    public static final String DESCRIPTION_TAG = "DS";
     public static final String DATE_RUN_PRODUCED_TAG = "DT";
     public static final String FLOW_ORDER_TAG = "FO";
     public static final String KEY_SEQUENCE_TAG = "KS";
-    public static final String DESCRIPTION_TAG = "DS";
-    public static final String PLATFORM_UNIT_TAG = "PU";
-    public static final String SEQUENCING_CENTER_TAG = "CN";
-    public static final String PLATFORM_TAG = "PL";
     public static final String LIBRARY_TAG = "LB";
+    public static final String PROGRAM_GROUP = "PG";
```
Comment (mcovarr) [inline src/java/htsjdk/samtools/SAMReadGroupRecord.java]: Why are all the others _TAG but PROGRAM_GROUP and PLATFORM_MODEL are not?

Comment (nh13) [inline src/java/htsjdk/samtools/SAMReadGroupRecord.java]: fair point

File: src/java/htsjdk/samtools/SAMReadGroupRecord.java [hunk 4]
```diff
@@ -117,6 +120,12 @@ public void setPredictedMedianInsertSize(final Integer predictedMedianInsertSize
         setAttribute(PREDICTED_MEDIAN_INSERT_SIZE_TAG, (predictedMedianInsertSize == null? null: predictedMedianInsertSize.toString())); 
     }
 
+    public String getProgramGroup() { return getAttribute(PROGRAM_GROUP); }
+    public void setProgramGroup(final String programGroup) { setAttribute(PROGRAM_GROUP, programGroup);}
```
Comment (mcovarr) [inline src/java/htsjdk/samtools/SAMReadGroupRecord.java]: formatting of new setter code is a bit off

**Comment by mcovarr** [comment]
:+1:

**Pull request merged**